### ISSUE TITLE
add option to hide deprecated APIs in web view

### ISSUE THREAD

**Issue by Luckey-Elijah** [dart-lang/dartdoc#3921]
Example: When browsing `pub.dev/documentation/somepackage` I would like the ability to "hide" all deprecated APIs as developers would not be needing to read the APIs.

**Comment by srawlins** [comment]
Thanks for the suggestion! I would say that, until the API is no longer available, a developer may want to see the deprecated API, so that they can read how to stop using it; what the replacement is; what the old signature is vs what the replacement is; why it is being removed, etc.

**Comment by Luckey-Elijah** [comment]
Yes I agree. I think that the ability to view them should be priority. Just for those that know they don't need to see them, it would be nice to hide those unused APIs.

_A more specific use case_: [`flutter_riverpod`](https://pub.dev/documentation/flutter_riverpod/latest/flutter_riverpod/flutter_riverpod-library.html) is preparing for a "big version 3 release" so many APIs are marked as deprecated. From a simple `document.querySelectorAll('.deprecated').length;` query I can see that 69 elements (at the time of this comment) are marked as deprecated. Those entries are valuable _to those who need them_ but can be an "eye sore" or generally unhelpful for those that are not using API deprecated APIs; hence, the idea to toggle hide/showing them.

I currently can use

```js
document.querySelectorAll('.deprecated').forEach(function(el) {
   el.style.display = 'none'; // use null to revert
});
```

in the browser console to get an idea of how I would want it to look.

Hopefully, this is more helpful for what I had in mind.

**Comment by srawlins** [comment]
Ah I see. Yes this would be a cool feature. A little toggle.

**Comment by jonasfj** [comment]
I guess package authors could use the `/// @nodoc` annotation on all the deprecated items they don't want to show them. But yes, a toggle would be nice.
